kvbc/include/merkle_tree_db_adapter.h

```diff
@@ -48,14 +49,18 @@ namespace concord::kvbc::v2MerkleTree {
 // getLastReachableBlockId() + 1. Additionally, ReachableSTBlock <= getLatestBlockId().
 class DBAdapter : public IDbAdapter {
  public:
+  using NonProvableKeySet = std::unordered_set<Key>;
   // Unless explicitly turned off, the constructor will try to link the blockchain with any blocks in the temporary
   // state transfer chain. This is done so that the DBAdapter will operate correctly in case a crash or an abnormal
   // shutdown has occurred prior to startup (construction). Only a single DBAdapter instance should operate on a
   // database and access to all methods should be either done from a single thread or serialized via a mutex or another
   // mechanism. The constructor throws if an error occurs.
   // Note1: The passed DB client must be initialized beforehand.
   // Note2: The 'linkTempSTChain' parameter turns of chain linking and is used for testing/tooling purposes.
-  DBAdapter(const std::shared_ptr<concord::storage::IDBClient> &db, bool linkTempSTChain = true);
+  // Note3: The key provided via 'non_provable_key_set' parameter will be stored outside of the Merkle Tree.
+  DBAdapter(const std::shared_ptr<concord::storage::IDBClient> &db,
+            bool linkTempSTChain = true,
+            const NonProvableKeySet &non_provable_key_set = NonProvableKeySet{});
```

Comment:
Nit: Since this file uses camelCase, maybe we can name it `nonProvableKeySet`.

Reply:
Done